DataFormats/Detectors/TRD/include/DataFormatsTRD/RawDataStats.h

```diff
@@ -0,0 +1,130 @@
+                     TRDParsingTrackletPadRowIncreaseError,                    // subsequent padrow can not be less than previous one.
+                     TRDParsingTrackletColIncreaseError,                       // subsequent col can not be less than previous one
+                     TRDParsingTrackletNoTrackletEndMarker,                    // got to the end of the buffer with out finding a tracklet end marker.
+                     TRDParsingTrackletExitingNoTrackletEndMarker              // got to the end of the buffer exiting tracklet parsing with no tracklet end marker
+};
+
+//enumerations for the options, saves on having a long parameter list.
+enum OptionBits {
+  TRDByteSwapBit,
+  TRDVerboseBit,
+  TRDHeaderVerboseBit,
+  TRDDataVerboseBit,
+  TRDCompressedDataBit,
+  TRDFixDigitCorruptionBit,
+  TRDEnableTimeInfoBit,
+  TRDEnableStatsBit,
+  TRDIgnoreDigitHCHeaderBit,
+  TRDIgnoreTrackletHCHeaderBit,
+  TRDEnableRootOutputBit
+};
+
+class TRDDataCountersPerEvent
+{ //thisis on a per event basis
+ public:
+  //TODO this should go into a dpl message for catching by qc ?? I think.
+  uint64_t mTimeTaken;                        // time take to process an event (summed trackletparsing and digitparsing) parts not accounted for.
+  uint64_t mTimeTakenForDigits;               // time take to process tracklet data blocks [us].
+  uint64_t mTimeTakenForTracklets;            // time take to process digit data blocks [us].
+  uint64_t mDigitWordsRead;                   // digit words read in
+  uint64_t mDigitWordsSkipped;                // digit words skipped for various reasons.
+  uint64_t mTrackletWordsRead;                // tracklet words read in
+  uint64_t mTrackletWordsSkipped;             // tracklet words skipped for various reasons.
+  std::array<uint8_t, 1080> mLinkErrorFlag{}; //status of the error flags for this event, 8bit values from cru halfchamber header.
+};
+
+class TRDDataCountersPerTimeFrame
+{ //thisis on a per event basis
+ public:
+  std::array<uint32_t, 1080> mLinkNoData;                                   // Link had no data or was not present.
+  std::array<uint32_t, 1080> mLinkWords{};                                  //units of 256bits, read from the cru half chamber header
+  std::array<uint32_t, 1080> mLinkWordsRead{};                              // units of 32 bits the data words read before dumping or finishing
+  std::array<uint32_t, 1080> mLinkWordsDumped{};                            // units of 32 bits the data dumped due to some or other error
+  std::array<int64_t, o2::trd::constants::MAXMCMCOUNT> mLinkMCMsWithData{}; // and its corresponding volume of data.
+  std::array<uint32_t, constants::MAXMCMCOUNT> mMCMDigitCount{};
+  std::array<uint32_t, constants::MAXMCMCOUNT> mMCMTrackletCount{};
+  std::array<uint32_t, 30> mParsingErrors{};              // errors in parsing, indexed by enum above of ParsingErrors
```

Comment:
It looks like the total parsing errors could be rebuilt from the error by link. But the overhead is small, so we can leave it.

Reply:
at the moment mParsingErrorsByLink are not filled hence the mParsingErrors, but your point it valid yes.